plugin/parser/engine/pg/convert_test.go

```diff
@@ -289,6 +381,61 @@ func equalColumnList(expected []*ast.ColumnDef, actual []*ast.ColumnDef) bool {
 	return true
 }
 
+func equalStringList(expected []string, actual []string) bool {
+	if len(expected) != len(actual) {
+		return false
+	}
+
+	for i := range expected {
+		if expected[i] != actual[i] {
+			return false
+		}
+	}
+
+	return true
+}
+
+func equalForeign(expected *ast.ForeignDef, actual *ast.ForeignDef) bool {
+	if expected == nil && actual == nil {
+		return true
+	}
+
+	if expected == nil || actual == nil {
+		return false
+	}
+
+	equalTable := equalTableDef(expected.Table, actual.Table)
+	equalColumnList := equalStringList(expected.ColumnList, actual.ColumnList)
+
+	return equalTable && equalColumnList
+}
+
+func equalConstraint(expected *ast.ConstraintDef, actual *ast.ConstraintDef) bool {
+	equalType := (expected.Type == actual.Type)
+	equalName := (expected.Name == actual.Name)
+	equalKey := equalStringList(expected.KeyList, actual.KeyList)
+	equalForeign := equalForeign(expected.Foreign, actual.Foreign)
+
+	return equalType && equalName && equalKey && equalForeign
+}
+
+func equalConstraintList(expected []*ast.ConstraintDef, actual []*ast.ConstraintDef) bool {
```

Comment:
Would pretty.Diff() work?

https://pkg.go.dev/github.com/kr/pretty#Diff

Reply:
Thanks for the nice advice! It's useful.